### PR TITLE
Branches- C. Gutierrez

### PR DESCRIPTION
**Pull request by CEsGutierrez**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model is the template for the data stored in the database. It provides the Controller with labels which it may use to access that information. 
Describe in your own words what the Controller is doing in Rails | The controller accesses the information in the database, or modifies it. 
Describe in your own words what the View is doing in Rails | The view interacts with the user directly. It is the display 
Describe an edge-case controller test you wrote | I wrote a test to check the deletion of an invalid ID. This is an edge case because to get to the delete function, one would have to interact with the index, which would make it challenging to attempt to delete something with an invalid ID. (I think, although writing this I'm absolutely sure there's a way to do it)
What is the purpose of using strong params? (i.e. the params method in the controller) | To reduce typing, typos, human error, accessibility of data/privacy without interacting with instances of Task coordinated by the Controller. 
How are Rails migrations related to Rails models? | Rails migrations are how the schemata of the database are created, modified, deleted. The schemata are directly aligned with the Model and one reflects the structure / data that is available in the other. 
Describe one area of Rails that are still unclear on | I would like to learn more about how to systematically write tests. I feel that I spent a lot of time guessing at syntax rather than thoughtfully trying to break my code.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  
+  root :to => 'tasks#index', as: 'root'
+  
+  get '/tasks/:id/edit', to: 'tasks#edit', as: 'edit_task'
+  
+  patch '/tasks/:id', to: 'tasks#update'
```
Comment (jmaddox19) [inline config/routes.rb:8]: I assume you know this now and just forgot to remove this but these but come with the code on line 12

**Comment by jmaddox19** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes!
Answered comprehension questions | X
Successfully handles: Index, Show | X
Index & Show tests pass | X
Successfully handles: New, Create | X
New & Create tests pass | X
Successfully handles: Edit, Update | X
Tests for Edit & Update test valid & invalid task ids | X
Successfully handles: Destroy, Task Complete | X
Tests for Destroy & Task Complete include tests for valid and invalid task ids | X
Routes follow RESTful conventions | X
Uses named routes (like `_path`) | X
**Overall** | Great job!

**Comment by jmaddox19** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes!
Answered comprehension questions | X
Successfully handles: Index, Show | X
Index & Show tests pass | X
Successfully handles: New, Create | X
New & Create tests pass | X
Successfully handles: Edit, Update | X
Tests for Edit & Update test valid & invalid task ids | X
Successfully handles: Destroy, Task Complete | X
Tests for Destroy & Task Complete include tests for valid and invalid task ids | X
Routes follow RESTful conventions | X
Uses named routes (like `_path`) | X
**Overall** | Great job!
